Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -169,3 +170,218 @@ void Gfx::AntiAliasingPainter::draw_cubic_bezier_curve(const FloatPoint& control
+            pixel(x, ymin, alpha);
+            ymin += 1;
+        }
+    };
+
+    auto eight_pixel = [&](int x, int y, int alpha) {
+        pixel(x, y, alpha);
+        pixel(x, -y - 1, alpha);
+        pixel(-x - 1, -y - 1, alpha);
+        pixel(-x - 1, y, alpha);
+        pixel(y, x, alpha);
+        pixel(y, -x - 1, alpha);
+        pixel(-y - 1, -x - 1, alpha);
+        pixel(-y - 1, x, alpha);
+    };
+
+    while (i < q) {
+        predict();
+        minimize();
+        correct();
+        v_old = v;
+        v += delta_y;
+        if (v >= 0) {
+            /* Single pixel on perimeter */
+            eight_pixel(i, q, (v + v_old) >> 1);
+            fill(i, q - 1, -q, intensity);
+            fill(-i - 1, q - 1, -q, intensity);
+        } else {
+            /* Two pixels on perimeter */
+            v += subpixel_resoulution;
+            eight_pixel(i, q, v_old >> 1);
+            q -= 1;
+            fill(i, q - 1, -q, intensity);
+            fill(-i - 1, q - 1, -q, intensity);
+            if (i < q) {
+                /* Haven't gone below the diagonal */
+                eight_pixel(i, q, (v + subpixel_resoulution) >> 1);
+                fill(q, i - 1, -i, intensity);
+                fill(-q - 1, i - 1, -i, intensity);
+            } else {
+                /* Went below the diagonal, fix v for final pixels */
+                v += subpixel_resoulution;
+            }
+        }
+        i += 1;
+    }
+
+    /* Fill in 4 remaning pixels */
+    int alpha = v >> 1;
+    pixel(q, q, alpha);
+    pixel(-q - 1, q, alpha);
+    pixel(-q - 1, -q - 1, alpha);
+    pixel(q, -q - 1, alpha);
+}
+
+void Gfx::AntiAliasingPainter::fill_rect_with_rounded_corners(IntRect const& a_rect, Color color, int radius)
+{
+    fill_rect_with_rounded_corners(a_rect, color, radius, radius, radius, radius);
+}
+
+void Gfx::AntiAliasingPainter::fill_rect_with_rounded_corners(IntRect const& a_rect, Color color, int top_left_radius, int top_right_radius, int bottom_right_radius, int bottom_left_radius)
+{
+    if (!top_left_radius && !top_right_radius && !bottom_right_radius && !bottom_left_radius)
+        return m_underlying_painter.fill_rect(a_rect, color);
+
+    if (color.alpha() == 0)
+        return;
+
+    IntPoint top_left_corner {
+        a_rect.x() + top_left_radius,
+        a_rect.y() + top_left_radius,
+    };
+    IntPoint top_right_corner {
+        a_rect.x() + a_rect.width() - top_right_radius,
+        a_rect.y() + top_right_radius,
+    };
+    IntPoint bottom_right_corner {
+        a_rect.x() + bottom_left_radius,
+        a_rect.y() + a_rect.height() - bottom_right_radius
+    };
+    IntPoint bottom_left_corner {
+        a_rect.x() + a_rect.width() - bottom_left_radius,
+        a_rect.y() + a_rect.height() - bottom_left_radius
+    };
+
+    IntRect top_rect {
+        a_rect.x() + top_left_radius,
+        a_rect.y(),
+        a_rect.width() - top_left_radius - top_right_radius, top_left_radius
```

Comment:
you have two fields on the same line here, while the others are on separate lines. Let's be consistent.